src/main/java/com/uber/kafkaSpraynozzle/KafkaSpraynozzle.java

```diff
@@ -42,13 +47,39 @@ public static void main(String[] args) {
             System.err.println(e.getMessage());
             return;
         }
+
+        String zk = spraynozzleArgs.getZk();
+        String leaderLatchZkPath = spraynozzleArgs.getLeaderLatchZkPath();
+        String counterZkPath = spraynozzleArgs.getDistributedAtomicCounterZkPath();
+        boolean buffering = spraynozzleArgs.getBuffering();
+        String hostName = getHostNameFromInetAddress();
+        if (isStringEmpty(hostName)) {
+            hostName = getHostNameFromFile();
+        }
+
+        if (!isStringEmpty(leaderLatchZkPath) && !isStringEmpty(counterZkPath)) {
+            System.out.println("Performing leader election through zookeeper and picking leader that will proceed.");
+            //use same zk as kafka
+            SpraynozzleLeaderLatch curatorClient = new SpraynozzleLeaderLatch(zk, leaderLatchZkPath, counterZkPath, "spraynozzle-" + hostName);
```

Comment:
Can we include the topic name and cleaned URL, too? There may be more than one spraynozzle sitting on the same box, though part of a different cluster.